doc/src/manual/style-guide.md

```diff
@@ -164,6 +164,62 @@ uses (e.g. `a[i]::Int`) than to try to pack many alternatives into one type.
 If a function name requires multiple words, consider whether it might represent more than one
 concept and might be better split into pieces.
 
+## Write functions that use similar argument precedence as Julia's `base/`
+
+As a general rule, the Base library uses the following order of arguments to functions,
+as applicable:
+
+1. **Function argument**.
+   This permits the use of `do` blocks for passing multiline anonymous functions.
+
+2. **I/O stream**.
+   Specifying the `IO` object first permits passing the function to functions such as
+   [`sprint`](@ref), e.g. `sprint(show, x)`.
+
+3. **Input being mutated**.
+   In [`fill!(x, v)`](@ref fill!), `x` is the object being mutated, and it appears before
+   the value to be inserted into `x`.
+
+4. **Type**.
+   Passing a type typically means that the output will have the given type.
+   In [`read(io, String)`](@ref read), the `IO` argument appears before the type.
+
+5. **Input not being mutated**.
+   In `fill!(x, v)`, `v` is *not* being mutated and it comes after `x`.
+
+6. **Key**.
+   For associative collections, this is the key of the key-value pair(s).
+   For other indexed collections, this is the index.
+
+7. **Value**.
+   For associative collections, this is the value of the key-value pair(s).
+   In cases like `fill!(x, v)`, this is `v`.
+
+8. **Everything else**.
+   Any other arguments.
+
+9. **Varargs**.
+   This refers to arguments that can be listed indefinitely at the end of a function call.
+   For example, in `Matrix{T}(uninitialized, dims)`, the dimensions can be given as a
+   `Tuple`, e.g. `Vector{T}(uninitialized, (1,2))`, or as `Varargs`, e.g.
+   `Vector{T}(uninitialized, 1, 2)`.
+
+10. **Keyword arguments**.
+   In Julia keyword arguments have to come last anyway; they're listed here for the sake
```

Comment:
They only have to come last in the function declaration. They can appear anywhere in the function call:
```jl
julia> f(x, y; kw1=1, kw2="hi") = (x, y, kw1, kw2)
f (generic function with 1 method)

julia> f(kw1="five", :foo, 9)
(:foo, 9, "five", "hi")

julia> f(kw1="five", :foo, kw2=12, 9)
(:foo, 9, "five", 12)
```

Reply:
Whoa, nifty. I didn't know that.